src/backend/opengl/CommandBufferGL.cpp

```diff
@@ -446,85 +374,249 @@ namespace backend { namespace opengl {
 
+                case Command::SetComputePipeline: {
+                    SetComputePipelineCmd* cmd = mCommands.NextCommand<SetComputePipelineCmd>();
+                    lastPipeline = ToBackend(cmd->pipeline).Get();
+
+                    lastPipeline->ApplyNow();
+                    pushConstants.OnSetPipeline(lastPipeline);
+                } break;
+
+                case Command::SetPushConstants: {
+                    SetPushConstantsCmd* cmd = mCommands.NextCommand<SetPushConstantsCmd>();
+                    uint32_t* data = mCommands.NextData<uint32_t>(cmd->count);
+                    pushConstants.OnSetPushConstants(cmd->stages, cmd->count, cmd->offset, data);
+                } break;
+
+                case Command::SetBindGroup: {
+                    SetBindGroupCmd* cmd = mCommands.NextCommand<SetBindGroupCmd>();
+                    ApplyBindGroup(cmd->index, cmd->group.Get(),
+                                   ToBackend(lastPipeline->GetLayout()), lastPipeline);
+                } break;
+
+                default: { UNREACHABLE(); } break;
+            }
+        }
+
+        // EndComputePass should have been called
+        UNREACHABLE();
+    }
+
+    void CommandBuffer::ExecuteRenderPass(RenderPassDescriptorBase* renderPass) {
+        GLuint fbo = 0;
+
+        // Create the framebuffer used for this render pass and calls the correct glDrawBuffers
+        {
+            // TODO(kainino@chromium.org): This is added to possibly work around an issue seen on
+            // Windows/Intel. It should break any feedback loop before the clears, even if there
+            // shouldn't be any negative effects from this. Investigate whether it's actually
+            // needed.
+            glBindFramebuffer(GL_READ_FRAMEBUFFER, 0);
+            // TODO(kainino@chromium.org): possible future optimization: create these framebuffers
+            // at Framebuffer build time (or maybe CommandBuffer build time) so they don't have to
+            // be created and destroyed at draw time.
+            glGenFramebuffers(1, &fbo);
+            glBindFramebuffer(GL_DRAW_FRAMEBUFFER, fbo);
+
+            // Mapping from attachmentSlot to GL framebuffer attachment points. Defaults to zero
+            // (GL_NONE).
+            std::array<GLenum, kMaxColorAttachments> drawBuffers = {};
+
+            // Construct GL framebuffer
+
+            unsigned int attachmentCount = 0;
+            for (uint32_t i : IterateBitSet(renderPass->GetColorAttachmentMask())) {
+                TextureViewBase* textureView = renderPass->GetColorAttachment(i).view.Get();
+                GLuint texture = ToBackend(textureView->GetTexture())->GetHandle();
+
+                // Attach color buffers.
+                glFramebufferTexture2D(GL_DRAW_FRAMEBUFFER, GL_COLOR_ATTACHMENT0 + i, GL_TEXTURE_2D,
+                                       texture, 0);
+                drawBuffers[i] = GL_COLOR_ATTACHMENT0 + i;
+                attachmentCount = i + 1;
```

Comment:
Nit: not new, but doing this every iteration seems redundant. Not sure if this works with IterateBitSet(), but scope i outside the loop and use it as attachmentCount after?

Reply:
Left as-is because we didn't decide yet wether render pass attachments have to be dense in [0, N] or can be sparse: I thought of using `renderPass->GetColorAttachmentMask().count()` but this might not be correct if render pass attachments are sparse.